### DISCUSSION TITLE
Limit branches/tags fetched on git sync to avoid memory usage

### DISCUSSION THREAD

**Discussion by alecgorge** [arxanas/git-branchless#1143] [General]
We use branches for release management at my company so we end up having 100s of tags/branches being created every day which makes `git sync` very slow. I'm not sure if this behavior is intentional for some reason I don't understand, but generally I only want to fetch the main branch and any of my branches I have locally tracking one on origin. 

It also causes immense memory usage (32+ GB in here, but I've seen it over 70GB before):
![CleanShot 2023-12-01 at 11 26 25@2x](https://github.com/arxanas/git-branchless/assets/261668/7c849510-bb6e-4731-8274-755f1a9d002e)
![CleanShot 2023-12-01 at 11 27 18@2x](https://github.com/arxanas/git-branchless/assets/261668/f400f17a-b597-4662-ae34-7a9adf5d019e)

**Comment by arxanas** [comment]
I think I've posted about this somewhere else, but can't find it now. The problem is that `git fetch` is invoked with `--all` during `git sync` to get the latest ref information. This is entirely because it was much easier from an implementation perspective to fetch them all, instead of selecting only the appropriate refs to fetch.

It would be unequivocally better to run `git fetch <remote1> <ref1> <ref2>...` for each remote/ref pair. Feel free to open a PR implementing this.

In the meantime, you can probably work around it by setting a [refspec](https://git-scm.com/book/en/v2/Git-Internals-The-Refspec) to fetch just the branches you want.
